test/hotspot/jtreg/compiler/vectorization/TestRoundVectorDoubleRandom.java

```diff
@@ -0,0 +1,156 @@
+import compiler.lib.ir_framework.TestFramework;
+import compiler.lib.ir_framework.Warmup;
+
+public class TestRoundVectorDoubleRandom {
+  private static final int ITERS  = 11000;
+  private static final int ARRLEN = 997;
+  private static final double ADD_INIT = -7500.;
+
+  private static final double[] input = new double[ARRLEN];
+  private static final long [] res = new long[ARRLEN];
+  private static final Random rand = new Random();
+
+  public static void main(String args[]) {
+    TestFramework.runWithFlags("-XX:-TieredCompilation", "-XX:CompileThresholdScaling=0.3", "-XX:MaxVectorSize=16");
+    TestFramework.runWithFlags("-XX:-TieredCompilation", "-XX:CompileThresholdScaling=0.3", "-XX:MaxVectorSize=32");
+  }
+
+  @DontCompile
+  long golden_round(double d) {
+    return Math.round(d);
+  }
+
+  @Test
+  @IR(counts = {IRNode.ROUND_VD, "> 0"},
+      applyIfPlatform = {"x64", "true"},
+      applyIfCPUFeature = {"avx512dq", "true"})
+  @IR(counts = {IRNode.ROUND_VD, "> 0"},
+      applyIfPlatform = {"aarch64", "true"})
+  void test_round(long[] a0, double[] a1) {
+    for (int i = 0; i < a0.length; i+=1) {
+      a0[i] = Math.round(a1[i]);
+    }
+  }
+
+  @Run(test = "test_round")
+  @Warmup(ITERS)
+  void test_rounds(RunInfo runInfo) {
+    // Initialize
+    for (int i = 0; i < ARRLEN; i++) {
+      double val = ADD_INIT+(double)i;
+      input[i] = val;
+    }
+
+    test_round(res, input);
+    // skip test/verify when warming up
+    if (runInfo.isWarmUp()) {
+      return;
+    }
+
+    int errn = 0;
+    // a double precise float point is composed of 3 parts: sign/e(exponent)/f(signicant)
+    // e (exponent) part of a float value
+    final int eShift = 52;
+    final int eWidth = 11;
+    final int eBound = 1 << eWidth;
+    // f (significant) part of a float value
+    final int fWidth = eShift;
+    final long fBound = 1 << fWidth;
+    final int fNum = 256;
+
+    // prepare for data of f (i.e. significand part)
+    long fis[] = new long[fNum];
+    int fidx = 0;
+    for (; fidx < fWidth; fidx++) {
+      fis[fidx] = 1 << fidx;
+    }
+    for (; fidx < fNum; fidx++) {
+      fis[fidx] = rand.nextLong(fBound);
+    }
+    fis[rand.nextInt(fidx)] = 0;
+
+    // generate input arrays for testing, then run tests & verify results
+    for (long fi : fis) {
+      // generate test input by combining different parts:
+      //   previously generated f values,
+      //   random value in e (i.e. exponent) range,
+      //   both positive and negative of previous combined values (e+f)
+      final int eStart = rand.nextInt(9);
+      final int eStep = (1 << 3) + rand.nextInt(3);
+      for (int ei = eStart; ei < eBound; ei += eStep) {
+        int eiIdx = ei/eStep;
+        // combine e and f
+        long bits = (ei << eShift) + fi;
+        // combine sign(+/-) with e and f
+        // positive values
+        input[eiIdx*2] = Double.longBitsToDouble(bits);
+        // negative values
+        bits = bits | (1 << 63);
```

Comment:
```suggestion
        bits = bits | (1L << 63);
```
And yet another of these

Reply:
```
jshell> 1 << 63
$4 ==> -2147483648
```